x-pack/plugins/monitoring/server/lib/calculate_auto.ts

```diff
@@ -25,18 +27,22 @@ const roundingRules = [
   [d(3, 'week'), d(1, 'week')],
   [d(1, 'year'), d(1, 'month')],
   [Infinity, d(1, 'year')],
-];
+] as RoundingRule[];
 
-function find(rules, check) {
-  function pick(buckets, duration) {
-    const target = duration / buckets;
+function find(
+  rules: RoundingRule[],
+  check: (b: number | Duration, i: Duration, t: number) => Duration | void
+) {
+  function pick(buckets?: number, duration?: Duration): Duration {
+    if (!buckets || !duration) return moment.duration(0);
+    const target = duration.asMilliseconds() / buckets;
     let lastResp;
 
     for (let i = 0; i < rules.length; i++) {
       const rule = rules[i];
       const resp = check(rule[0], rule[1], target);
 
-      if (resp == null) {
+      if (resp === null || resp === undefined) {
```

Comment:
This caught my attention (maybe there's a check return type that would avoid the conditional?) and I found myself confused by what these functions are doing. There's only one test case which doesn't clear it up for me either. Probably not an item for this PR but if you know what it does it might be good to get some comments in here.

If you don't that's cool too. I see that it's fairly old code.

Reply:
All I know is that when I changed it to `resp === null` it failed because `resp` was `undefined`. I think that happens when the `check` tries to `pick` an `interval` that doesn't exist.

Anyway this is at least more explicit about what this line of code is doing. `==` is inherently not type safe.